jetstream/external_config.py

```diff
@@ -39,6 +44,99 @@ def validate(self, experiment: jetstream.experimenter.Experiment) -> None:
         Analysis("no project", "no dataset", conf).validate()
 
 
+def validate_config_settings(config_file: Path) -> None:
+    """
+    Implemented to resolve a Github issue:
+        - https://github.com/mozilla/jetstream/issues/843
+
+    Loads external config file and runs a number of validation steps on it:
+    - Checks that all config keys/settings are lowercase
+    - Checks for missing core config keys
+    - Checks for unexpected core configuration keys
+    - Checks that all segments defined under experiment have configuration in segments section
+    - Checks if metric with custom config is defined in metrics.weekly or metrics.overall fields
+
+    Returns None, if issues found with the configuration an Exception is raised
+    """
+
+    config = toml.loads(config_file.read_text())
+
+    optional_core_config_keys = (
+        "metrics",
+        "experiment",
+        "segments",
+        "data_sources",
+        "friendly_name",
+        "description",
+    )  # TODO: double check that friendly_name and description are to be expected here
+
+    core_config_keys_specified = config.keys()
+
+    # checks for unexpected core configuration keys
+    if unexpected_config_keys := (set(core_config_keys_specified) - set(optional_core_config_keys)):
+        err_msg = f"Unexpected config key[s] found: {unexpected_config_keys}"
+        raise UnexpectedKeyConfigurationException(err_msg)
+
+    # checks that all segments defined under experiment have configuration in segments section
+    if experiment_config := config.get("experiment", dict()):
+        expected_segment_configuration = experiment_config.get("segments", list())
+
+        if expected_segment_configuration:
+            segments_config_keys = config.get("segments", dict()).keys()
+
+            # TODO: will desktop include all possible segments?
```

Comment:
Can we do something similar like we are doing here? https://github.com/mozilla/jetstream/blob/4ca2d3e217dde6722ccc63cf40255c71aa5ded32/jetstream/config.py#L94-L102
```suggestion
```

Reply:
I tweaked it a bit, I still need to introduce additional couple of lines as `__all__` does not provide exactly the info I needed.